### PR TITLE
ci: Detect unused code with Periphery

### PR DESCRIPTION
**Pull request by PhilippeWeidmann**
Unused code is now detected using periphery. (https://github.com/peripheryapp/periphery)
For every PR a comment is generated which gives a list of unused code + the count.

### CODE REVIEW

File: scripts/periphery.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+eval "$($HOME/.local/bin/mise activate -C $SRCROOT bash --shims)"
+
+tuist install
+tuist generate -n
```
Comment (adrien-coye) [inline scripts/periphery.sh:6]: yup -n makes sense

File: scripts/periphery.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+# Check if the "--full-dir" argument is provided
+if [[ "$1" == "--full-dir" ]]; then
+    full_dir=true
+else
+    full_dir=false
+fi
+
+# Perform the periphery scan and handle directory removal based on argument
+if "$full_dir"; then
```
Comment (adrien-coye) [inline scripts/periphery.sh:17]: Defining a bool just to if else on it, we could probably put everything in the first parameter check.

**Review by adrien-coye — APPROVED** [review]
Looks good to me. Checkout my comment about defining an extra variable, you can either change it or merge this as is. As you prefer.

File: scripts/periphery.sh (hunk 2 above)
Comment (PhilippeWeidmann) [reply scripts/periphery.sh]: yup totally agree

**Review by valentinperignon — APPROVED** [review]

**Pull request merged**